### ISSUE TITLE
Intermittent timeout in /old-tests/submission/Opera/script_scheduling/109.html

### ISSUE THREAD

**Issue by jdm** [servo/servo#7019]
Bad run:
```
10:03.98 TEST_START: Thread-TestrunnerManager-1 /old-tests/submission/Opera/script_scheduling/109.html
10:04.12 PROCESS_OUTPUT: Thread-TestrunnerManager-1 (pid:78549) Full command: /Users/servo/buildbot/slave/mac1/build/target/release/servo --cpu --hard-fail -u Servo/wptrunner -z http://localhost:8000/old-tests/submission/Opera/script_scheduling/109.html
(pid:78549) "ERROR:ALERT: RESULT: {"tests":[{"name":" scheduler: javascript URL in iframe, src set via DOM","status":1,"message":"assert_array_equals: lengths differ, expected 5 got 3","stack":"onload<@http://localhost:8000/old-tests/submission/Opera/script_scheduling/109.html:15:7\nTest.prototype.step@http://localhost:8000/resources/testharness.js:1379:20\nTest.prototype.step_func/<@http://localhost:8000/resources/testharness.js:1403:1\n"}],"status":0,"message":null,"stack":null}"
10:04.13 PROCESS_OUTPUT: Thread-TestrunnerManager-1 (pid:78549) "js::rust: Error at :0: TypeError: top.log is not a function"
10:04.13 PROCESS_OUTPUT: Thread-TestrunnerManager-1 (pid:78549) ""
10:19.00 TEST_END: Thread-TestrunnerManager-1 TIMEOUT, expected OK
```
Good run:
```
10:04.59 TEST_START: Thread-TestrunnerManager-2 /old-tests/submission/Opera/script_scheduling/109.html
10:04.74 PROCESS_OUTPUT: Thread-TestrunnerManager-2 (pid:55077) Full command: /Users/servo/buildbot/slave/mac1/build/target/release/servo --cpu --hard-fail -u Servo/wptrunner -z http://localhost:8000/old-tests/submission/Opera/script_scheduling/109.html
(pid:55077) "ERROR:js::rust: Error at :0: TypeError: top.log is not a function"
10:04.74 PROCESS_OUTPUT: Thread-TestrunnerManager-2 (pid:55077) ""
10:04.78 TEST_END: Thread-TestrunnerManager-2 Harness OK. Subtests passed 0/1. Unexpected 0
```

**Comment by jdm** [comment]
This looks like https://github.com/servo/rust-mozjs/blob/master/src/rust.rs#L725 needs a similar treatment to #6917.

**Comment by jdm** [comment]
Might get fixed by #7027.

**Comment by nox** [comment]
This isn't disabled and #7027 landed a long time ago.
